nemo/core/classes/modelPT.py

```diff
@@ -60,6 +62,8 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
 
             trainer (Optional): Pytorch Lightning Trainer instance
         """
+        if is_dataclass(cfg) and isinstance(cfg, ModelPTConfig):
```

Comment:
Only line really needed to support Data classes. After this, they are treated as OmegaConf / DictConfig object so no other code changes necessary. Serialization and deserialization is unaffected beyond this point.